include/wil/registry_helpers.h

```diff
@@ -140,30 +182,52 @@ namespace wil
                 }
 
                 ::std::vector<::std::wstring> strings;
-                auto current = first;
-                const auto end_iterator = last;
-                const auto last_null = (end_iterator - 1);
-                while (current != end_iterator)
-                {
-                    const auto next = ::std::find(current, end_iterator, L'\0');
-                    if (next != end_iterator)
+                walk_multistring(first, last, [&](const InputIt& string_first, const InputIt& string_last)
                     {
-                        if (next != last_null)
-                        {
-                            // don't add an empty string for the final 2nd-null-terminator
-                            strings.emplace_back(current, next);
-                        }
-                        current = next + 1;
-                    }
-                    else
-                    {
-                        current = next;
+                        strings.emplace_back(string_first, string_last);
                     }
-                }
+                );
                 return strings;
             }
 #endif // #if defined(_VECTOR_) && defined(_STRING_) && defined(WIL_ENABLE_EXCEPTIONS)
 
+            template <size_t C>
+            void get_multistring_bytearray_from_strings_nothrow(const PCWSTR data[C], ::wil::unique_cotaskmem_array_ptr<BYTE>& multistring) WI_NOEXCEPT

```

Comment:
because the calling pattern is just passing through the caller's wil container. like

::wil::reg::reg_view_details::get_cotaskmemstring_array_from_multistring_nothrow(begin, end, return_value);

Reply:
Oh, it's an implicit _inout_ here. Gotchu!